### PR TITLE
PRD-148: Add JKS compatibility for cert exporter

### PR DESCRIPTION
**Pull request by Yifan-Wu-99**
JKS expiration metrics are flowing to grafana in [here](https://teletracking.grafana.net/explore?schemaVersion=1&panes=%7B%22gTf%22:%7B%22datasource%22:%22grafanacloud-teletrackingent-prom%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22cert_exporter_secret_expires_in_seconds%7Bkey_name%3D%5C%22jks%5C%22%7D%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22grafanacloud-teletrackingent-prom%22%7D,%22editorMode%22:%22builder%22,%22legendFormat%22:%22__auto%22,%22useBackend%22:false,%22disableTextWrap%22:false,%22fullMetaSearch%22:false,%22includeNullMetadata%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D%7D%7D&orgId=1)

### CODE REVIEW

**Comment by dean-allen-tele** [comment]
Looks like there is no unit tests on this repo. I think we should add some for the functionality you are adding.

**Comment by Yifan-Wu-99** [comment]
> Looks like there is no unit tests on this repo. I think we should add some for the functionality you are adding.

@dean-allen-tele  There are tests for testing getting metrics for certs that live in k8s under the test folder, which requires defining a certificate with all info there. I don't want to paste there our own certificate(secret) for testing because it's public, just validated and tested in our own cluster and it worked, so it should be good.

File: src/exporters/certHelpers.go [hunk 1]
```diff
@@ -49,7 +51,13 @@ func secondsToExpiryFromCertAsBytes(certBytes []byte, password string) ([]certMe
 	if parsed {
 		return metrics, nil
 	}
-	return nil, fmt.Errorf("failed to parse as pem and pkcs12: %w", err)
+	// Parse as JKS
+	parsed, metrics, err = parseAsJKS(certBytes, password)
+	if parsed {
+		glog.Infof("JKS file parsing successful!")
```
Comment (andrew-qu2000) [inline src/exporters/certHelpers.go]: You could add this log for the other types (PEM, PKCS) to be consistent. Maybe lower the log level to debug/trace though cause there will be a lot of secrets.

File: src/exporters/certHelpers.go [hunk 2]
```diff
@@ -117,3 +125,21 @@ func parseAsPEM(certBytes []byte) (bool, []certMetric, error) {
...
+// parseAkJKS parses a Java Key Store (JKS) certificate file to extract the certificates and their metrics
+func parseAsJKS(certBytes []byte, password string) (bool, []certMetric, error) {
+	var metrics []certMetric
+
+	jks, err := jks.Parse(certBytes, &jks.Options{Password: password})
+	if err != nil {
+		glog.Errorf("Failed to parse as a jks: %v", err)
+		return false, nil, err
+	}
+	certChain := jks.Keypairs[0].CertChain
```
Comment (andrew-qu2000) [inline src/exporters/certHelpers.go:137]: a comment about what Keypairs[0] is could be useful

**Pull request merged**